rskj-core/src/test/java/co/rsk/net/NodeBlockProcessorTest.java

```diff
@@ -1082,4 +1084,66 @@ void failIfProcessBlockHeadersRequestCountHigher()  {
         verify(sender, never()).sendMessage(any());
 
     }
+
+    @Test
+    void processNewBlockHashesMessageOnlyRequestsNonAdvancedBlock() {
+        final NetBlockStore store = new NetBlockStore();
+        final Peer sender = mock(SimplePeer.class);
+        final BlockNodeInformation nodeInformation = mock(BlockNodeInformation.class);
+        final SyncConfiguration syncConfiguration = SyncConfiguration.IMMEDIATE_FOR_TESTING;
+        final Blockchain blockchain = new BlockChainBuilder().ofSize(0);
+
+        TestSystemProperties config = new TestSystemProperties();
+        BlockSyncService blockSyncService = new BlockSyncService(config, store, blockchain, nodeInformation, syncConfiguration, DummyBlockValidator.VALID_RESULT_INSTANCE);
+        final NodeBlockProcessor processor = new NodeBlockProcessor(store, blockchain, nodeInformation, blockSyncService, syncConfiguration);
+
+        final long advancedBlockNumber = (long) syncConfiguration.getChunkSize() * syncConfiguration.getMaxSkeletonChunks() + blockchain.getBestBlock().getNumber() + 1;
+
+        BlockIdentifier advancedBlockIdentifier = new BlockIdentifier(new byte[32], advancedBlockNumber);
+        BlockIdentifier nonAdvancedBlockIdentifier = new BlockIdentifier(new byte[32], advancedBlockNumber - 1);
+
+        // create list of block identifiers with advanced and non-advanced blocks
+        List<BlockIdentifier> blockIdentifierList = new ArrayList<>();
+
+        blockIdentifierList.add(nonAdvancedBlockIdentifier);
+        blockIdentifierList.add(advancedBlockIdentifier);
+
+        NewBlockHashesMessage message = new NewBlockHashesMessage(blockIdentifierList);
+
+        processor.processNewBlockHashesMessage(sender, message);
+
+        // verify that only the non-advanced block is requested and added to the node information
+        verify(sender, times(1)).sendMessage(any(GetBlockMessage.class));
+        verify(nodeInformation, times(1)).addBlockToNode(any(), any());
```

Comment:
ArgumentCaptor could be used here and check if the block added is the right one, as into this test is not truly checked. Having that maybe the next unit test wouldn't be needed.

Reply:
That's a pretty useful idea. Just Implemented it and now with one test is enough. Thanks!!!